ios/RNSpokestack.swift

```diff
@@ -286,25 +352,33 @@ class RNSpokestack: RCTEventEmitter, SpokestackDelegate {
 
         // Initialize TTS
         synthesizer = TextToSpeech([self], configuration: speechConfig)
+        
+        // Set resolve now in case
+        // all downloads are synchronous, early returns
+        // from the cache and the last one builds the pipeline
+        resolvers[RNSpokestackPromise.initialize] = resolve
+        rejecters[RNSpokestackPromise.initialize] = reject
 
-        if nluFiles == 3 {
-            do {
-                try classifier = NLUTensorflow([self], configuration: speechConfig)
-            } catch {
-                handleError(error)
-                reject("init_error", "Spokestack NLU initialization failed", error)
+        // Set to total before starting requests
+        // in case the downloader returns the cached version synchronously.
+        // This avoids wakeword building the pipeline before moving on to NLU
+        numRequests = (wakeDownloads.count == 3 ? 3 : 0) + (nluDownloads.count == 3 ? 3 : 0)
```

Comment:
given that you're also looking for the models by name elsewhere, it's probably worth sticking them into a `String`-valued enum that you can then just do the `count` on here and following

Reply:
The number of requests needs to be tracked separately and dynamically. I don't follow the enum strategy.